rviz_common/src/rviz_common/temp/resource_retriever/retriever.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2009, Willow Garage, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the names of Stanford University or Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef RESOURCE_RETRIEVER_RETRIEVER_H
+#define RESOURCE_RETRIEVER_RETRIEVER_H
+
+#include <stdint.h>
+#include <string>
+#include <stdexcept>
+
+typedef void CURL;
+
+namespace resource_retriever
+{
+
+class Exception : public std::runtime_error
+{
+public:
+  Exception(const std::string& file, const std::string& error_msg)
+  : std::runtime_error("Error retrieving file [" + file + "]: " + error_msg)
+  {}
+};
+
+/**
+ * \brief A combination of a pointer to data in memory along with the data's size.
+ */
+struct MemoryResource
+{
+  MemoryResource()
+  : size(0)
+  {}
+
+  std::shared_ptr<uint8_t> data;
```

Comment:
Import `<memory>` when using `std::shared_ptr`.

Reply:
I'll add it, but this file (since it is in the `temp` folder) will probably be deleted in the future.

Reply:
I know, but the error would persist if the file is moved instead.